Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs

```diff
@@ -173,13 +152,52 @@ protected override void Dispose(bool disposing)
 				if (_drawerLayout != null && _drawerListener != null)
 				{
 					_drawerLayout.RemoveDrawerListener(_drawerListener);
```

Comment:
The issue is `_drawerLayout` is disposed because of the above `foreach` loop. This block of code should be executed at the end so that we can remove drawer listeners safely before disposing the MDP.